### ISSUE TITLE
quality-suite METRIC_OFF: multinomial_logit_recovers_true_softmax_and_beats_statsmodels — log-loss 1.12624 exceeds oracle 1.06701 by >0.05 (==statsmodels)

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#699]
## Summary

The quality-suite test `multinomial_logit_recovers_true_softmax_and_beats_statsmodels` (in `tests/quality_vs_statsmodels_multinomial.rs`, ~line 96) fails its **PRIMARY** near-Bayes-optimal log-loss check on the synthetic arm. gam's held-out log-loss is **1.12624**, exceeding the oracle (true-softmax / Bayes-optimal) log-loss **1.06701** by **0.05923 nats/obs**, just past the 0.05 slack. Decisively, **gam and statsmodels MNLogit match each other exactly** (`rel_l2_vs_sm = 0.00000`, both = 1.12624). The two independent maximum-likelihood multinomial fits agree to floating point, so the gap is finite-sample parameter noise at `n_train = 200`, not a gam likelihood defect.

## Failing assertion & observed values

```rust
// tests/quality_vs_statsmodels_multinomial.rs:325
assert!(
    gam_log_loss <= oracle_log_loss + 0.05,
    "gam held-out log-loss {gam_log_loss:.5} exceeds Bayes-optimal \
     {oracle_log_loss:.5} by more than 0.05 nats/obs"
);
```

Observed: `gam held-out log-loss 1.12624 exceeds Bayes-optimal 1.06701 by more than 0.05 nats/obs` (gap = 0.05923 > 0.05).

The match-or-beat-statsmodels check (`gam_log_loss <= sm_log_loss + 0.005`, line 335) **passes**: gam == statsmodels at 1.12624 (`rel_l2 = 0.00000`).

## Mathematical background

**K-class softmax.** With `K` classes and class `K-1` as the reference (`η_{K-1} ≡ 0`), the model probabilities for covariate row `x` are

$$p_k(x) = \frac{\exp(\eta_k(x))}{\sum_{j=0}^{K-1}\exp(\eta_j(x))}, \qquad k = 0,\dots,K-1.$$

Labels are **sampled** (not argmax'd) from the true softmax `p^{\text{true}}_k(x_i)` built from `true_coef`, so the data-generating distribution is explicitly known.

**Oracle / Bayes-optimal log-loss.** For a held-out set, the cross-entropy log-loss of probability matrix `q` against observed labels `y` is

$$\mathrm{LL}(q) = -\frac{1}{n}\sum_{i=1}^{n}\sum_{k=0}^{K-1}\mathbf 1\{y_i=k\}\,\ln q_k(x_i).$$

The oracle plugs in the TRUE softmax `q = p^{\text{true}}`. Its population expectation is the conditional entropy (Bayes risk),

$$\mathbb E[\mathrm{LL}(p^{\text{true}})] = \mathbb E_x\Big[-\sum_{k} p^{\text{true}}_k(x)\,\ln p^{\text{true}}_k(x)\Big],$$

the irreducible minimum any model can attain. The **observed** oracle value 1.06701 is the finite-sample (`n_test`) Monte-Carlo realization of this Bayes risk on the actual sampled labels.

**Why a perfect estimator still overshoots by >0.05.** A model fitted on `n_train` rows estimates `η̂` with covariance `≈ I(θ)^{-1}/n_train`. To second order, the expected excess log-loss over the oracle is

$$\mathbb E[\mathrm{LL}(\hat p) - \mathrm{LL}(p^{\text{true}})] \approx \frac{1}{2n_{\text{train}}}\,\mathrm{tr}\big(I^{-1}I_{\text{test}}\big) \;=\; O\!\Big(\frac{d}{n_{\text{train}}}\Big),$$

with `d` the number of free softmax parameters (`(K-1)×(p+1)`). At `n_train = 200` this Akaike-style excess is non-negligible and is incurred by **any** consistent estimator — including the exact MLE that statsmodels also computes. The 0.0592 observed gap is this finite-sample term.

## Analysis — suspected root cause

gam and statsmodels MNLogit are two independent multinomial MLEs and they **agree to floating point** (`rel_l2 = 0.00000`, identical 1.12624 log-loss). Therefore gam's likelihood, softmax inversion, and reference-class gauge are all correct — there is nothing gam could change to close the gap without overfitting. The 0.05-nat oracle slack is simply **smaller than the irreducible `O(d/n_train)` estimation excess** at `n_train = 200`: parameter noise alone pushes the best achievable held-out log-loss ~0.059 above the oracle. The gate is over-tight for this sample size.

## Classification

**Over-tight oracle-gap gate (finite-sample parameter noise).** Not a gam-vs-statsmodels mismatch (they're identical), not a likelihood defect. Principled fixes: widen the oracle slack to reflect the `(K-1)(p+1)/n_train` excess (or scale it with `d/n_train`), increase `n_train`, or gate the gam-vs-statsmodels agreement (already passing) plus an absolute log-loss ceiling rather than the tight oracle slack.

## Related issues

- #453, #409 (multinomial).

## Reproduction

```
cargo test --test quality_vs_statsmodels_multinomial multinomial_logit_recovers_true_softmax_and_beats_statsmodels
cargo test --test quality_vs_statsmodels_multinomial multinomial_logit_recovers_true_softmax_and_beats_statsmodels_on_real_data
```

_Filed from the reference-quality-suite METRIC_OFF review._

**Comment by SauersML** [comment]
## Verdict: FIXED (gate already relaxed in source) — pending build to RUN

### Root cause
The original gate `gam_log_loss <= oracle_log_loss + 0.05` was **below the achievable finite-sample optimum** at `n_train=200`: with sampled (not argmax) labels the unpenalized multinomial MLE has genuine parameter variance, so even a perfect fit sits ~0.06 nats above the Bayes-optimal oracle. The reported `gam_logloss=1.12624` vs `oracle=1.06701` is a 0.0592-nat gap that is pure irreducible MLE variance, not a fitter defect — a prior investigation showed gam == statsmodels to float (`prob rel_l2 = 0.00000`) and statsmodels lands at the same ~1.126.

### Current source (verified via gh api)
`tests/quality_vs_statsmodels_multinomial.rs`:
- line 335: `gam_log_loss <= oracle_log_loss + 0.10` (widened from +0.05).
- line 345: `gam_log_loss <= sm_log_loss + 0.005` (the real objective: match-or-beat statsmodels).

With gap = 0.0592 < 0.10 the absolute clause **passes**, and gam==sm so the match-or-beat clause **passes**. The +0.10 widening is present in **both** the 0.1.170 release commit (382ba3d11) and current main, so this is already resolved at source.

### Current state (empirical) / gap
This test exercises the internal `fit_penalized_multinomial` (zero-penalty MLE) Rust API, not a Python entry point, so I can only RUN it via `cargo test` on a checkout. The numbers above are byte-deterministic (the test uses a fixed LCG), so the pass is robust. Recommend `cargo test --release gam_multinomial_logit_recovers_true_softmax_and_beats_statsmodels` to close. No code change needed.

**Comment by SauersML** [comment]
Closing — **verified PASS** on current main in the Reference Quality Suite CI run [27079010667](https://github.com/SauersML/gam/actions/runs/27079010667) (commit 28dcd2c1d, which includes the relevant fixes). The named test `quality_vs_statsmodels_multinomial` now passes (outcome=PASS). Empirically confirmed on Linux CI, not source-only.
